### ISSUE TITLE
Abandoned Carts Issue

### ISSUE THREAD

**Issue by Tangol** [villagedefrance/OpenCart-Overclocked#172]
I've noticed that when you click on the **View** abandoned cart button, it takes you to the **Order Details** page.
However, the abandoned cart is not really an order.

I'm getting the logic of this, as the customer may email/call the support team and ask them to go ahead with the order on his behalf, so it's easier to just process it from there.

- The problem is that after you do this (process the order on the customer's behalf), the cart doesn't get updated and the products still remain there.

Also, I'm a bit worried about the order **IDs** from the abandoned carts and the ones from the real orders, conflicting.
Is there any separation between them and can you please elaborate on how the system works?

- And finally, we need a contact counter for the abandoned cart reminder emails sent to customers.
A nice way of doing this, would be like in the image below.
Also, the green checkmark needs a `title` attribute saying **Reminder Sent**.
![contacted](https://user-images.githubusercontent.com/3336728/38206551-8094d6a4-36b3-11e8-8960-0a6ada2cb9fe.jpg)

Tested on https://github.com/villagedefrance/OpenCart-Overclocked/commit/fbacaa2a81a5711f7005b13b54896d9bb77d8992

**Comment by villagedefrance** [comment]
I've just made an update on this.
Now, regarding the email counter, I don't see that as necessary since only 1 single email should be sent anyway, with only 2 possible outcomes: First, the customer wants to recover the order. Second, the customer does not want the order, then in this case, the abandoned order should be deleted.
My latest updates should make the whole process much clearer now.

**Comment by Tangol** [comment]
The new system works fine, with one exception: _after you recover the order, the cart doesn't get updated in the front end and the products still remain there._

**Comment by villagedefrance** [comment]
I have been thinking about this and also tried a few things, but I came to the conclusion that the front end Cart should not be cleared. And this is why:

- the technical limitations: the shopping Cart is a Session based feature and is for this reason quite volatile. Each time you re-connect, a new session is created and old sessions cannot be recovered, Carts and Wishlists are re-loaded onto the new session. The Admin session is also distinct from the Catalog sessions, so exchanging data between the two sessions is near impossible. That's why we can't change Products, Shipping and Payment when recovering an abandoned order, because the new session would break it.

- the time frame: the time between the Cart being abandoned and the time this Cart is being recovered could be days or possibly weeks. In this time the customer has maybe updated is Cart with new products. So deleting it could compromise future orders.

- the missed attempt: The customer could have bailed out of the first order for a number of reasons: pressed the wrong button, forgot to add something else, received a phone call at that time, ...
So he/she maybe managed to placed a valid order later and by doing so already cleared the Cart.

- the follow-up: If an Abandoned Cart email was sent and the customer realizes that is order hasn't gone through, then maybe he/she is going to try to place it again. The item is still in the Cart at that time but will then be cleared once the valid order is processed.

- more scenarios?

What do you think?

**Comment by Tangol** [comment]
I agree with all your points, however in my report, I was referring to only one issue, and that is the clearing of the Cart when the order is being recovered by the the **Admin** at the customer's request.

For example, we have the following instance:
- Customer ads some products to the Cart, then something happens and he has to abandon the order.
- Having completely forgotten about it, a few days later he receives an email regarding the abandoned cart.
Seeing all the contents of the cart directly in the email, and being a busy person, he decides to directly call the store to tell them to go ahead with the order on his behalf.
- The store admin recovers the cart and processes the order.
- Following this step, the cart should be cleared, as the order got placed by the Admin, on the customer's behalf.

Right now after the order recovery, the cart doesn't get cleared, and if the customer visits the site again, he will get confused to whether or not the order was placed at his request.

So I think we should find a way to clear the Cart only when the order gets recovered by the admin.

**PS: By order recovery I mean:**
- Send reminder to customer
- View the abandoned order
- Click on **Recover Order**
- Verify the requested data and the final step (**Totals**), change the status of the order to whatever.
- Click **Save** and voilà, the order is placed.
So when the **Admin** changes the **Order Status** and clicks **Save**, at this point the cart should be cleared.

Hope it makes sense.

**Comment by villagedefrance** [comment]
I understood that the first time but like I said before we are talking about Sessions for this matter, and because of the nature of Sessions, we can really do that safely from Admin without consequences.

The best solution here, if not clearing the Cart is really an issue for some users, is to login as the customer through the Customer's List under Admin, therefore open a new customer session, and clear its Cart manually this way, removing only what needs to be removed. Any other way should be considered unreliable.

**Comment by Tangol** [comment]
Everything is clear now.
I have nothing else to add, except the minor suggestion above:

> Also, if the customer calls back, it would be nice if we could search the abandoned carts, using his name.
> This is very important for large stores, that have dozens of abandoned carts.

As far as I'm concerned, the issue can be closed.
Thanks!
